drivers/sensor/ite_vcmp_it8xxx2/vcmp_ite_it8xxx2.c

```diff
@@ -0,0 +1,398 @@
+/*
+ * Copyright (c) 2022 ITE Technology Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT ite_it8xxx2_vcmp
+
+#include <device.h>
+#include <devicetree/io-channels.h>
+#include <drivers/adc.h>
+#include <drivers/sensor.h>
+#include <drivers/sensor/it8xxx2_vcmp.h>
+#include <dt-bindings/dt-util.h>
+#include <dt-bindings/sensor/it8xxx2_vcmp.h>
+#include <errno.h>
+#include <soc.h>
+#include <soc_dt.h>
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(vcmp_ite_it8xxx2, CONFIG_SENSOR_LOG_LEVEL);
+
+#define VCMP_REG_MASK		0x7
+#define VCMP_RESOLUTION		BIT(10)
+#define VCMP_MAX_MVOLT		3000
+
+/* Device config */
+struct vcmp_it8xxx2_config {
+	/* Voltage comparator x control register */
+	uintptr_t reg_vcmpxctl;
+	/* Voltage comparator x channel select MSB register */
+	uintptr_t reg_vcmpxcselm;
+	/* Voltage comparator scan period register */
+	uintptr_t reg_vcmpscp;
+	/* Voltage comparator x threshold data buffer MSB register */
+	uintptr_t reg_vcmpxthrdatm;
+	/* Voltage comparator x threshold data buffer LSB register */
+	uintptr_t reg_vcmpxthrdatl;
+	/* Voltage comparator status register */
+	uintptr_t reg_vcmpsts;
+	/* Voltage comparator status 2 register */
+	uintptr_t reg_vcmpsts2;
+	/* Voltage comparator module irq */
+	int irq;
+	/* Voltage comparator channel */
+	int vcmp_ch;
+	/* Scan period for "all voltage comparator channel" */
+	int scan_period;
+	/*
+	 * Determines the condition between ADC data and threshold_mv
+	 * that will trigger voltage comparator interrupt.
+	 */
+	int comparison;
+	/* Threshold assert value in mv */
+	int threshold_mv;
+	/* Pointer of ADC device that will be performing measurement */
+	const struct device *adc;
+};
+
+/* Driver data */
+struct vcmp_it8xxx2_data {
+	/* ADC channel config */
+	struct adc_channel_cfg adc_ch_cfg;
+	/* Work queue to be notified when threshold assertion happens */
+	struct k_work work;
+	/* Sensor trigger hanlder to notify user of assetion */
+	sensor_trigger_handler_t handler;
+	/* Pointer of voltage comparator device */
+	const struct device *vcmp;
+};
+
+/* Voltage comparator work queue address */
+static uint32_t vcmp_work_addr[VCMP_CHANNEL_CNT];
```

Comment:
This looks suspicious, does it belong in `struct vcmp_it8xx2_data`?

Reply:
When the interrupt fire, we need to check all the comparator channels status and handle the triggered callback function. Each vcmp channel could has its work queue data in vcmp_it8xxx2_data_##inst, but we need all the channels work queue in interrupt function.